client/asset/eth/multirpc.go

```diff
@@ -1096,18 +1100,12 @@ func (m *multiRPCClient) withAll(
 	if atLeastOne {
 		return nil
 	}
-	if errs == nil {
-		return errors.New("all providers in a failed state")
-	}
-	// TODO: use errors.Join(errs) when Go 1.20 is the min
-	var b []byte
-	for i, err := range errs {
-		if i > 0 {
-			b = append(b, '\n')
-		}
-		b = append(b, err.Error()...)
+
+	if len(errs) == len(providers) {
+		return fmt.Errorf("all providers in a failed state: %s", strings.Join(errs, "\n"))
```

Comment:
You can probably just do `if len(errs) > 0 {` here, no?

Reply:
"all providers" won't be correct with `len(errs) > 0` check. We will only ever get here if `atLeastOne` is false and if so,  `len(errs)` must  equal  `len(providers)`

Reply:
RE: https://github.com/decred/dcrdex/pull/2803#discussion_r1649048918

If we are reporting errors, which we should, len(err) can't be `nil` hence the refactored check.

Reply:
What happens if all providers have `p.failed() == true`?